### PR TITLE
Cleanup CFG_'s generic configurations

### PR DESCRIPTION
**Pull request by etienne-lms**
`CFG_xxx` are configuration directive and should not be set from the C/Asm source files.

`CFG_TEE_RAM_xxx` and `CFG_TA_RAM_xxx` are never set from the env and very few platforms offer `CFG_SHMEM_xxx`. This change removes `CFG_` prefix from these as well as `TEE_LOAD_ADDR`.

However, platforms can allows some configurations to be taken into account from platform_config.h, i.e:
```c
#ifdef CFG_TEE_LOAD_ADDR
#define TEE_LOAD_ADDR      CFG_TEE_LOAD_ADDR
#else
#define TEE_LOAD_ADDR      TEE_RAM_START
#endif
```
This change does not modify the platform implementation, it is (should be) only about renaming.

This change is the first step for removing all `CFG_xxx` definitions from **platform_config.h** files, as proposed by https://github.com/OP-TEE/optee_os/pull/2273.

### CODE REVIEW

**Comment by etienne-lms** [comment]
If that's too messy, i can remove the SDP commit out of this p-r.

**Comment by etienne-lms** [comment]
travis failed on checkpatch. Will fix.

File: core/arch/arm/plat-zynq7k/platform_config.h [hunk 1]
```diff
@@ -212,11 +212,11 @@
...
 #define TZDRAM_BASE			(CFG_DDR_TEETZ_RESERVED_START)
 #define TZDRAM_SIZE				(CFG_DDR_TEETZ_RESERVED_SIZE - \
-						CFG_PUB_RAM_SIZE)
+							CFG_PUB_RAM_SIZE)
 
-#define CFG_SHMEM_START			(CFG_DDR_TEETZ_RESERVED_START + \
+#define TEE_SHMEM_START			(CFG_DDR_TEETZ_RESERVED_START + \
 						TZDRAM_SIZE)
-#define CFG_SHMEM_SIZE			CFG_PUB_RAM_SIZE
+#define TEE_SHMEM_SIZE			CFG_PUB_RAM_SIZE
```
Comment (jenswi-linaro) [inline core/arch/arm/plat-zynq7k/platform_config.h]: We should consider replacing `CFG_PUB_RAM_SIZE` with `CFG_SHMEM_SIZE`, doesn't have to be done in this commit though.

File: documentation/porting_guidelines.md [hunk 2]
```diff
@@ -143,33 +143,27 @@ could look like this:
...
 
-#define DRAM0_BASE		0x00000000
-#define DRAM0_SIZE		0x40000000
-
-/* Below ARM-TF */
-#define TEE_SHMEM_START		0x08000000
-#define TEE_SHMEM_SIZE		(4 * 1024 * 1024)
-
-/* If your device has SRAM */
+/* Optional: When used with CFG_WTI_PAGER, define the device SRAM */
```
Comment (jenswi-linaro) [inline documentation/porting_guidelines.md]: CFG_WITH_PAGER

**Comment by jenswi-linaro** [comment]
Drop the
```
Conflicts:
        core/arch/arm/plat-imx/config/config_imx6multi.h
        core/arch/arm/plat-imx/config/config_imx6ul.h
```
lines in `"core: remove CFG_ prefix from CFG_TEE_LOAD_ADDR"`
With that `"core: remove CFG_ prefix from *"` commits:
`Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>`

File: core/arch/arm/plat-ti/platform_config.h [hunk 3]
```diff
@@ -131,19 +131,26 @@
 
 #endif /* CFG_WITH_PAGER */
 
+/* Locate SDP memory at the end of TZDRAM: caller sets CFG_TEE_SDP_MEM_SIZE */
```
[the remark this answers is not in the document]
Comment (etienne-lms) [reply core/arch/arm/plat-ti/platform_config.h]: thanks. I will replace `caller sets CFG_...` with `configuration directives set CFG_...`.
(in less than 80 chars :(

File: core/arch/arm/plat-ti/platform_config.h [hunk 4]
```diff
@@ -131,19 +131,26 @@
...
+#ifndef CFG_TEE_SDP_MEM_SIZE
+#error "Requires CFG_TEE_SDP_MEM_SIZE"
```
[the remark this answers is not in the document]
Comment (etienne-lms) [reply core/arch/arm/plat-ti/platform_config.h]: oups! thanks.
Actually, it is useless. If `CFG_TEE_SDP_MEM_SIZE` build should complain for an undefined reference.
I will remove it.

File: core/arch/arm/plat-zynq7k/platform_config.h (hunk 1 above)
Comment (etienne-lms) [reply core/arch/arm/plat-zynq7k/platform_config.h]: ok. Whatever, this part will need some rework. There is equivalent changes on going through https://github.com/OP-TEE/optee_os/pull/2217

File: documentation/porting_guidelines.md (hunk 2 above)
Comment (etienne-lms) [reply documentation/porting_guidelines.md]: ok, thanks

File: documentation/porting_guidelines.md [hunk 5]
```diff
@@ -143,33 +143,28 @@ could look like this:
...
 
-#define DRAM0_BASE		0x00000000
-#define DRAM0_SIZE		0x40000000
-
-/* Below ARM-TF */
-#define CFG_SHMEM_START		0x08000000
-#define CFG_SHMEM_SIZE		(4 * 1024 * 1024)
-
-/* If your device has SRAM */
+/* Optional: When used with CFG_WTI_PAGER, define the device SRAM */
```
Comment (etienne-lms) [inline documentation/porting_guidelines.md]: will fix

**Comment by jenswi-linaro** [comment]
Since you started this despite #2217 I guess you've already answered the question.

**Comment by etienne-lms** [comment]
my bad. I completely forgot #2217 when starting this. This raised from #2261.
Well, I guess this part is generic. It is more the 2nd half (platforms update to get rid of `#define CFG_xxx` from **platform_config.h** files) that really conflict with #2217 changes. So, let's go for this one.

**Comment by etienne-lms** [comment]
Review tag applied to commits "core: remove CFG_ prefix from *".

Rebased on master, and add a commit to prevent checkpatch issues when modifying poplar **platform_config.h**.

File: core/arch/arm/plat-marvell/platform_config.h [hunk 6]
```diff
@@ -154,14 +147,28 @@
...
-					  CFG_TEE_SDP_MEM_TEST_SIZE, \
+					  TEE_SDP_TEST_MEM_SIZE, \
 					  CORE_MMU_DEVICE_SIZE)
 
-/* Secure data path test memory pool: located at end of TA RAM */
-#if CFG_TEE_SDP_MEM_TEST_SIZE
-#define CFG_TEE_SDP_MEM_SIZE		CFG_TEE_SDP_MEM_TEST_SIZE
-#define CFG_TEE_SDP_MEM_BASE		(TZDRAM_BASE + TZDRAM_SIZE - \
-						CFG_TEE_SDP_MEM_SIZE)
+/*
```
[the remark this answers is not in the document]
Comment (etienne-lms) [reply core/arch/arm/plat-marvell/platform_config.h]: I guess. Same with the generic way of defining `TEE_RAM_*` and `TA_RAM_*` from `TZDRAM_*` / `TZSRAM_*` and when platforms emulate SRAM using DRAM.
Let's see that afterward if you agree.

[the remark this answers is not in the document]
Comment (jenswi-linaro) [reply core/arch/arm/plat-marvell/platform_config.h]: OK

**Comment by jenswi-linaro** [comment]
`Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Pull request merged**